.travis.yml

```diff
@@ -19,19 +19,35 @@ matrix:
       before_install: source script/travis/install_qt5.sh
     - os: linux
       env: STATIC_ANALYSIS=ON
+    - os: linux
+      env: DYNAMIC_ANALYSIS=ON CMAKE_OPTIONS="-DCMAKE_BUILD_TYPE=Debug"
+      addons:
+        apt:
+          update: true
+          packages: valgrind
     - os: osx
       env: QT_BASE=latest CMAKE_OPTIONS="-DPENGUINV_BUILD_QT_EXAMPLE=ON"
       before_install: source script/travis/install_qt5.sh
 
 script:
-  - if [[ "$STATIC_ANALYSIS" == "ON" ]]; then
+  - |
+    if [[ "$STATIC_ANALYSIS" == "ON" ]]; then
       export EXCLUDE_PATTERN_1=" not found in expected header ";
       export EXCLUDE_PATTERN_2=" does not need to be \#included; use a forward declaration instead";
       export EXCLUDE_PATTERN_3=" not found in any directly \#included header";
       git clone https://github.com/myint/cppclean;
       ./cppclean/cppclean src test examples | grep -v "$EXCLUDE_PATTERN_1" | grep -v "$EXCLUDE_PATTERN_2" | grep -v "$EXCLUDE_PATTERN_3";
       exit 0;
     fi
+  - |
+    if [[ "$DYNAMIC_ANALYSIS" == "ON" ]]; then
+      mkdir -p build && cd build
+      cmake .. ${CMAKE_OPTIONS}
+      cmake --build . --target unit_tests -- -j2
+      ctest -D ExperimentalBuild -j2
+      ctest -E perf_test --extra-verbose -T MemCheck -j2
+      exit 0
```

Comment:
It could be that this line causes us to dot not execute `after_script`. Try to remove it and see what happens.